### PR TITLE
feat!: adopt the new multi-pattern resource names

### PR DESCRIPTION
**Pull request by hzyi-google**
- Delete `TenantOrProjectName`.
- Generate overloads that take `TenantName` and `ProjectName` respectively where `TenantOrProjectName` was originally used.
- Delete per-pattern resource name helper classes: `CompanyWithTenantName`, `CompanyWithoutTenantName`, `JobWithTenantName`, `JobWithoutTenantName`, `CompanyOldName`, `CompanyNewName`, `JobOldName`, `JobNewName`.
- Delete resource name factory classes `JobNames`, `CompanyNames`, `TenantOrProjectNames` and untyped resource name classes `UntypedJobName`, `UntypedTenantName` and `UntypedTenantOrProjectName`.

### CODE REVIEW

**Comment by hzyi-google** [comment]
PTAL @chingor13

File: proto-google-cloud-talent-v4beta1/clirr-ignored-differences.xml [hunk 1]
```diff
@@ -16,4 +16,13 @@
     <className>com/google/cloud/talent/v4beta1/*OrBuilder</className>
     <method>boolean has*(*)</method>
   </difference>
+  <difference>
```
Comment (chingor13) [inline proto-google-cloud-talent-v4beta1/clirr-ignored-differences.xml]: Can we add the note to remove this after 0.36.0?

File: proto-google-cloud-talent-v4beta1/clirr-ignored-differences.xml [hunk 2]
```diff
@@ -16,4 +16,13 @@
...
+    <differenceType>8001</differenceType>
+    <className>com/google/cloud/talent/v4beta1/*Name*</className>
```
Comment (chingor13) [inline proto-google-cloud-talent-v4beta1/clirr-ignored-differences.xml]: Let's be more explicit on exactly which classes are ok to remove in this PR.

File: google-cloud-talent/clirr-ignored-differences.xml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- see http://www.mojohaus.org/clirr-maven-plugin/examples/ignored-differences.html -->
+<differences>
+  <!--TODO: To be removed-->
```
Comment (chingor13) [inline google-cloud-talent/clirr-ignored-differences.xml]: Please add after 0.36.0

File: proto-google-cloud-talent-v4beta1/clirr-ignored-differences.xml (hunk 1 above)
Comment (hzyi-google) [reply proto-google-cloud-talent-v4beta1/clirr-ignored-differences.xml]: done

File: proto-google-cloud-talent-v4beta1/clirr-ignored-differences.xml (hunk 2 above)
Comment (hzyi-google) [reply proto-google-cloud-talent-v4beta1/clirr-ignored-differences.xml]: done

File: google-cloud-talent/clirr-ignored-differences.xml (hunk 3 above)
Comment (hzyi-google) [reply google-cloud-talent/clirr-ignored-differences.xml]: done

**Pull request merged**